xarray/tests/test_units.py

```diff
@@ -1245,6 +1311,766 @@ def test_dot_dataarray(dtype):
+                ),
+            ),
+            pytest.param(unit_registry.m, id="identical_unit"),
+        ),
+    )
+    @pytest.mark.parametrize(
+        "convert_data",
+        (
+            pytest.param(False, id="no_conversion"),
+            pytest.param(True, id="with_conversion"),
+        ),
+    )
+    @pytest.mark.parametrize("func", (method("equals"), method("identical")), ids=repr)
+    def test_comparisons(self, func, unit, convert_data, dtype):
+        array = np.linspace(0, 1, 9).astype(dtype)
+        quantity1 = array * unit_registry.m
+        variable = xr.Variable("x", quantity1)
+
+        if convert_data and is_compatible(unit_registry.m, unit):
+            quantity2 = convert_units(array * unit_registry.m, {None: unit})
+        else:
+            quantity2 = array * unit
+        other = xr.Variable("x", quantity2)
+
+        expected = func(
+            strip_units(variable),
+            strip_units(
+                convert_units(other, extract_units(variable))
+                if is_compatible(unit_registry.m, unit)
+                else other
+            ),
+        )
+        if func.name == "identical":
+            expected &= extract_units(variable) == extract_units(other)
+        else:
+            expected &= all(
+                compatible_mappings(
+                    extract_units(variable), extract_units(other)
+                ).values()
+            )
+
+        actual = func(variable, other)
+
+        assert expected == actual
+
+    @pytest.mark.parametrize(
+        "unit",
+        (
+            pytest.param(1, id="no_unit"),
+            pytest.param(unit_registry.dimensionless, id="dimensionless"),
+            pytest.param(unit_registry.s, id="incompatible_unit"),
+            pytest.param(unit_registry.cm, id="compatible_unit"),
+            pytest.param(unit_registry.m, id="identical_unit"),
+        ),
+    )
+    def test_broadcast_equals(self, unit, dtype):
+        base_unit = unit_registry.m
+        left_array = np.ones(shape=(2, 2), dtype=dtype) * base_unit
+        value = (
+            (1 * base_unit).to(unit).magnitude if is_compatible(unit, base_unit) else 1
+        )
+        right_array = np.full(shape=(2,), fill_value=value, dtype=dtype) * unit
+
+        left = xr.Variable(("x", "y"), left_array)
+        right = xr.Variable("x", right_array)
+
+        units = {
+            **extract_units(left),
+            **({} if is_compatible(unit, base_unit) else {None: None}),
+        }
+        expected = strip_units(left).broadcast_equals(
+            strip_units(convert_units(right, units))
+        ) & is_compatible(unit, base_unit)
+        actual = left.broadcast_equals(right)
+
+        assert expected == actual
+
+    @pytest.mark.parametrize(
+        "indices",
+        (
+            pytest.param(4, id="single index"),
+            pytest.param([5, 2, 9, 1], id="multiple indices"),
```

Comment:
I think `Variable` supports more complicated indexing. This could be added later

Reply:
probably, though we just want to make sure the units survive through the indexing.